### PR TITLE
Added SKLearn-like random forest API.

### PR DESCRIPTION
**Pull request by canonizer**

### CODE REVIEW

File: dask_xgboost/core.py [hunk 1]
```diff
@@ -303,6 +303,19 @@ def predict(client, model, data):
     return result
 
 
+def _xgb_regressor_fit(obj, X, y):
```
Comment (mtjrider) [inline dask_xgboost/core.py]: Methods should be fully encapsulated within a class object unless they are utilities useful for core operation.

File: dask_xgboost/core.py [hunk 2]
```diff
@@ -303,6 +303,19 @@ def predict(client, model, data):
...
+                          num_boost_round=obj.get_num_boosting_rounds())
+    return obj
+
+
+def _xgb_regressor_predict(obj, X):
+    client = default_client()
+    return predict(client, obj._Booster, X)
+
+
 class XGBRegressor(xgb.XGBRegressor):
```
Comment (mtjrider) [inline dask_xgboost/core.py]: I think it makes the most sense to have a single class object called `XGBRegressor` which initializes, in a self-consistent way, both the code path for RF regression, and standard boosted regression.

File: dask_xgboost/core.py [hunk 3]
```diff
@@ -315,23 +328,100 @@ def fit(self, X, y=None):
...
+        ``eval_metric``, ``early_stopping_rounds`` and ``verbose`` fit
+        kwargs.
+        """
+        return _xgb_regressor_fit(self, X, y)
+
+    def predict(self, X):
+        return _xgb_regressor_predict(self, X)
+
+
+class XGBRFRegressor(xgb.XGBRFRegressor):
```
Comment (mtjrider) [inline dask_xgboost/core.py]: To unify the two class objects defined here, I think we have a few options:
```
class XGBRegressor(xgb.XGBRegressor, xgb.XGBRFRegressor):
  def __init__(self, RF=False):
    self.RF = RF # boolean flag to determine which regressor to use in Python
  def fit(self, ...):
    if(self.RF): ...
    ...
  def predict(self, ...):
    if(self.RF): ...
    ...

class XGBClassifier(xgb.XGBClassifier, xgb.XGBRFClassifier):
...
```

or something like:
```
class XGBRegressor:
  def __init__(self):
    self.XGBReg = xgb.XGBRegressor()
    self.XGBRFReg = xgb.XGBRFRegressor()
...
```

File: dask_xgboost/core.py [hunk 4]
```diff
@@ -303,6 +303,19 @@ def predict(client, model, data):
...
 
+def _xgb_regressor_fit(obj, X, y):
+    client = default_client()
+    xgb_options = obj.get_xgb_params()
+    obj._Booster = train(client, xgb_options, X, y,
+                          num_boost_round=obj.get_num_boosting_rounds())
+    return obj
+
+
+def _xgb_regressor_predict(obj, X):
```
Comment (mtjrider) [inline dask_xgboost/core.py]: In `master`, `def _*` methods are reserved for Dask utilities, and parent calls which we don't intend to expose to the user.

Method useful for a particular class object are encapsulated by that class, and generally not hidden.

**Review by mtjrider — CHANGES_REQUESTED** [review]

**Comment by canonizer** [comment]
I've updated the classes; they now use mixins and multiple inheritance.

**Review by mtjrider — APPROVED** [review]

**Pull request merged**